src/Abstraction/IModule.cs

```diff
@@ -14,7 +13,9 @@ public interface IModule
         IStructType StatType { get; }
         IStructType ThreadType { get; }
         IStructType VaListType { get; }
-        IEnumerable<IFunction> Functions { get; }
-        IEnumerable<IGlobal> Globals { get; }
+
+        IDefinition GetMain();
+        IFunction GetFunction(FunctionId functionId);
+        IGlobal GetGlobal(GlobalId globalId);
```

Comment:
FYI I have a trivial PR lined up where I shorten the names of these various id parameters to just id (everywhere in the solution).